project/src/components/video-player/video-player.tsx

```diff
@@ -0,0 +1,46 @@
+// Библиотеки
+import { useEffect, useRef } from 'react';
+
+type VideoPlayerProps = {
+  isPlaying: boolean;
+  muted: boolean;
+  src: string;
+  previewImage: string;
+  width: number;
+  height: number;
+}
+
+export default function VideoPlayer({ isPlaying, src, previewImage, muted, width, height }: VideoPlayerProps): JSX.Element {
+  // const setIsLoading = useState(true)[1];
+
+  const videoRef = useRef<HTMLVideoElement | null>(null);
+
+  useEffect(() => {
+    if (videoRef.current === null) {
+      return;
+    }
+
+    if (isPlaying) {
+      videoRef.current.play();
+      return;
+    }
+
+    videoRef.current.pause();
+    videoRef.current.currentTime = 0;
+    // setIsLoading(false);
+
+  }, [isPlaying]);
```

Comment:
Есть несколько способов сбросить видео в начальное состояние, например 
```suggestion
useEffect(() => {
    if (videoRef.current) {
      isPlaying ? videoRef.current.play() : videoRef.current.src = src;
    }
  }, [isPlaying, src]);
```
или еще один вариант вместо сброса src вызвать .load(), но по моему это приведет к новой загрузке видео с сервера, нужно проверять.